app/src/main/assets/po_token.html

```diff
@@ -0,0 +1,211 @@
+<!DOCTYPE html>
+<html lang="en"><head><title></title><script>
+class BotGuardClient {
+  constructor(options) {
+    this.userInteractionElement = options.userInteractionElement;
+    this.vm = options.globalObj[options.globalName];
+    this.program = options.program;
+    this.vmFunctions = {};
+    this.syncSnapshotFunction = null;
+  }
+
+  /**
+   * Factory method to create and load a BotGuardClient instance.
+   * @param options - Configuration options for the BotGuardClient.
+   * @returns A promise that resolves to a loaded BotGuardClient instance.
+   */
+  static async create(options) {
+    return await new BotGuardClient(options).load();
+  }
+
+  async load() {
+    if (!this.vm)
+      throw new Error('[BotGuardClient]: VM not found in the global object');
+
+    if (!this.vm.a)
+      throw new Error('[BotGuardClient]: Could not load program');
+
+    const vmFunctionsCallback = (
+      asyncSnapshotFunction,
+      shutdownFunction,
+      passEventFunction,
+      checkCameraFunction
+    ) => {
+      this.vmFunctions = {
+        asyncSnapshotFunction: asyncSnapshotFunction,
+        shutdownFunction: shutdownFunction,
+        passEventFunction: passEventFunction,
+        checkCameraFunction: checkCameraFunction
+      };
+    };
+
+    try {
+      this.syncSnapshotFunction = await this.vm.a(this.program, vmFunctionsCallback, true, this.userInteractionElement, () => {/** no-op */ }, [ [], [] ])[0];
+    } catch (error) {
+      throw new Error(`[BotGuardClient]: Failed to load program (${error.message})`);
+    }
+
+    // an asynchronous function runs in the background and it will eventually call
+    // `vmFunctionsCallback`, however we need to manually tell JavaScript to pass
+    // control to the things running in the background by interrupting this async
+    // function in any way, e.g. with a delay of 1ms. The loop is most probably not
+    // needed but is there just because.
+    for (let i = 0; i < 10000 && !this.vmFunctions.asyncSnapshotFunction; ++i) {
+      await new Promise(f => setTimeout(f, 1))
+    }
```

Comment:
I … don’t think this is how async works. The timeout is just gonna be scheduled on a new task, but the code before the loop still runs on a microtask on the previous task.

Reply:
The loop actually executes only once as far as I know, I still put a loop because you never know